sshd-common/src/test/java/org/apache/sshd/common/file/root/RootedFileSystemProviderTest.java

```diff
@@ -155,10 +174,15 @@ public void testMkdir() throws IOException {
         }
     }
 
-    @Test(expected = NoSuchFileException.class)
-    public void testMkdirInvalid() throws IOException {
-        Path parent = fileHelper.createDirectory(fileSystem.getPath(DOESNT_EXIST + getCurrentTestName()));
-        fail(String.format("Unexpected success in creating directory %s", parent.toString()));
+    @Test
+    public void testMkdirInvalid() {
+        if (isTestingRealWindowsHostFS) {
+            return;
+        }
```

Comment:
Better than just returning would be `Assume.assumeFalse(isTestingRealWindowsHostFS);`.

Bu could you explain why the tests have to be skipped on a real Windows FS?